### PR TITLE
Stop using Safari paths for SafariForWebKitDevelopment

### PR DESCRIPTION
**Pull request by szewai**
#### 84e59dc75f5e8c15934d1d3d2f91b33e89e60ba0
<pre>
Stop using Safari paths for SafariForWebKitDevelopment
<a href="https://bugs.webkit.org/show_bug.cgi?id=250063">https://bugs.webkit.org/show_bug.cgi?id=250063</a>
rdar://102422209

Reviewed by Chris Dumez.

SafariForWebKitDevelopment is used for testing local builds of WebKit, which can contain untrusted code, so it should
not touch Safari files / directories in data vaults (e.g. ~/Library/Safari).

* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::applicationOrProcessIdentifier):
(WebKit::defaultWebsiteDataStoreRootDirectory):
(WebKit::WebsiteDataStore::tempDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):

Canonical link: <a href="https://commits.webkit.org/258470@main">https://commits.webkit.org/258470@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/e10d20fb8a75b02eb4013d22b94bfcdcc9ed8db0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102026 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11170 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35095 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111355 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12141 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2088 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94425 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109104 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107807 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9298 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92565 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/37130 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91178 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24048 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/78854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4744 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25476 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4842 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1922 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10911 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44964 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5810 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6590 "Built successfully") | | | 
| | | | |

### CODE REVIEW

**Comment by aproskuryakov** [comment]
The overall approach of adjusting default paths in WebKit makes sense to me. Not quite clear why SafariForWebKitDevelopment needs to be special cased though, it feels like it could be no different from any arbitrary 3rd party app (say, MiniBrowser). Am I missing a reason for special casing it?

**Comment by szewai** [comment]
> The overall approach of adjusting default paths in WebKit makes sense to me. Not quite clear why SafariForWebKitDevelopment needs to be special cased though, it feels like it could be no different from any arbitrary 3rd party app (say, MiniBrowser). Am I missing a reason for special casing it?

SafariForWebKitDevelopment is using the same bundle identifier as Safari, and WebKit decides the default path using bundle identifier, so SafariForWebKitDevelopment uses paths like ~/Library/com.apple.Safari. The system has special protection on Safari's paths (as mentioned in the radar); it will throw sandbox error if the process is not whitelisted (e.g. having entitlement).

Other unsandboxed third-party app would use their own paths, e.g. ~/Library/org.webkit.MiniBrowser, which does not have such restriction. In this patch we would start to use ~/Library/SafariForWebKitDevelopment for SafariForWebKitDevelopment.

File: Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm [hunk 1]
```diff
@@ -317,7 +324,7 @@ static String defaultWebsiteDataStoreRootDirectory()
     }
 #endif
 
-    return cacheDirectoryFileSystemRepresentation("OfflineWebApplicationCache"_s, { }, ShouldCreateDirectory::No);
+    return cacheDirectoryFileSystemRepresentation("OfflineWebApplicationCacheTest"_s, { }, ShouldCreateDirectory::No);
```
Comment (cdumez) [inline Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm]: Huh?

Comment (szewai) [reply Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm]: Oops used for testing... will remove this part

File: Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm [hunk 2]
```diff
@@ -106,6 +106,17 @@ static bool experimentalFeatureEnabled(const String& key, bool defaultValue = fa
     return defaultValue;
 }
 
+static NSString* applicationOrProcessIdentifier()
+{
+    NSString *identifier = [NSBundle mainBundle].bundleIdentifier;
+    NSString *processName = [NSProcessInfo processInfo].processName;
+    if ([identifier isEqualToString:@"com.apple.Safari"] && [processName isEqualToString:@"SafariForWebKitDevelopment"])
```
Comment (cdumez) [inline Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:115]: Would be nice to add a comment to explain why we're special casing SafariForWebKitDevelopment here.

File: Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm (hunk 1 above)
Comment (cdumez) [reply Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm]: Should likely drop this change.

**Review by cdumez — APPROVED** [review]

**Pull request merged**